atomic_reactor/plugin.py

```diff
@@ -258,10 +284,21 @@ def run(self, keep_going=False):
                 logger.exception("failed to save plugin duration")
 
             self.plugins_results[plugin_class.key] = plugin_response
+
+            if plugin_successful and buildstep_phase:
+                logger.debug('stopping further execution of plugins '
+                             'after first successful plugin')
+                break
+
         if len(failed_msgs) == 1:
             raise PluginFailedException(failed_msgs[0])
         elif len(failed_msgs) > 1:
             raise PluginFailedException("Multiple plugins raised an exception: " + str(failed_msgs))
+
+        if not plugin_successful and buildstep_phase and not plugin_response:
```

Comment:
288 has plugin_successful True, so it won't hit here
but if all plugins are skipped OR plugin fails (no raising), I have no other way to distinguish those 2 cases then also check plugin_response